presto-main/src/main/java/io/prestosql/operator/SimplePageWithPositionEqualsAndHash.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.spi.Page;
+import io.prestosql.spi.block.Block;
+import io.prestosql.spi.type.Type;
+import io.prestosql.type.BlockTypeOperators;
+import io.prestosql.type.BlockTypeOperators.BlockPositionHashCode;
+import io.prestosql.type.BlockTypeOperators.BlockPositionIsDistinctFrom;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntList;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class SimplePageWithPositionEqualsAndHash
+        implements PageWithPositionEqualsAndHash
+{
+    private final IntList equalityChannels;
+    private final List<BlockPositionIsDistinctFrom> distinctFromOperators;
+    private final List<BlockPositionHashCode> hashOperators;
+
+    public SimplePageWithPositionEqualsAndHash(List<Type> channelTypes, List<Integer> equalityChannels, BlockTypeOperators blockTypeOperators)
+    {
+        requireNonNull(channelTypes, "channelTypes is null");
+        this.equalityChannels = new IntArrayList(requireNonNull(equalityChannels, "equalityChannels is null"));
+        checkArgument(channelTypes.size() >= equalityChannels.size(), "channelTypes cannot have fewer columns then equalityChannels");
+
+        // Use IS DISTINCT FROM for equality, because it evaluates NULL values as distinct (unlike SQL EQUALS).
```

Comment:
window operator behaves differently regarding `NaNs` (https://github.com/trinodb/trino/pull/6462). They are not equal there because window operator uses `io.prestosql.operator.PagesHashStrategy#rowEqualsRow` (which treats nulls as equal, but not `NaNs`)

I think we should use equality here too. It should be fine to have two rows non-equal but `compare=0`. Essentially, such rows will be placed next to each other in absolute ordering, but they will still form different peer groups

Reply:
For reference, had a discussion with @sopel39 on slack, and the correct SQL standard is to treat the rows using IS DISTINCT FROM as they are here. But we will maintain the prior incorrect behavior in this PR, to handle migration in another PR>